### PR TITLE
NEW: Add ViewableData::getViewerTemplates()

### PR DESCRIPTION
**Pull request by kinglozzer**
Forms part of the fix for silverstripe/silverstripe-cms#2039, merge before https://github.com/silverstripe/silverstripe-cms/pull/2041

### CODE REVIEW

**Comment by kinglozzer** [comment]
Uhh, so my editor decided to re-order those namespace imports... I can remove them if they’ll make merges a pain

File: src/ORM/DataObject.php [hunk 1]
```diff
@@ -2146,6 +2147,14 @@ public function getFrontEndFields($params = null)
         return $untabbedFields;
     }
 
+    /**
+     * {@inheritdoc}
+     */
```
Comment (robbieaverill) [inline src/ORM/DataObject.php]: AFAIK this would be the same result as not having a doc block at all, so may as well omit it

**Comment by kinglozzer** [comment]
I’ve pointed this to the wrong branch as the new API is required for https://github.com/silverstripe/silverstripe-cms/pull/2041 to be merged. So it’s kinda both a bugfix and new API... 😛

Rationale: https://github.com/silverstripe/silverstripe-cms/pull/2041#issuecomment-348520027

File: src/ORM/DataObject.php (hunk 1 above)
Comment (kinglozzer) [reply src/ORM/DataObject.php]: Good point, will remove 😄

File: src/View/ViewableData.php [hunk 2]
```diff
@@ -578,6 +579,17 @@ public function getIterator()
...
 
+    /**
+     * Find appropriate templates for SSViewer to use to render this object
+     *
+     * @param string $suffix
+     * @return array
+     */
+    public function getViewerTemplates($suffix = '')
+    {
+        return SSViewer::get_templates_by_class(static::class, $suffix, self::class);
```
Comment (tractorcow) [inline src/View/ViewableData.php]: 👍

**Comment by tractorcow** [comment]
We have a rather unique case here, where we want to add a new API to fix a bug. @kinglozzer feels it should be 4.0, @robbieaverill feels it should be 4. Personally I'm leaning towards 4 as well, but I want to invite others to include their view on this.

Any of the other @silverstripe/core-team want to vote on whether this is 4.0 or 4? 👍  this comment for 4.0, 👎  for 4.

**Comment by robbieaverill** [comment]
We're as early as possible in the 4.x release cycle, so now would be the time to do this in 4.0 if we need to. This change looks super useful.

**Comment by sminnee** [comment]
If it's code that supports a bugfix then I think it's okay. If you wanted to be really strict you could mark it as "internal" in 4.0 and remove that when you merge to 4.

**Pull request merged**